tests/integration/steps/integration_steps.go

```diff
@@ -0,0 +1,622 @@
+	backupChartPath, err := filepath.Abs("../../charts/backup")
+	g(t).Expect(err).To(gomega.BeNil())
+
+	helmOptions := &helm.Options{
+		SetValues: map[string]string{
+			"name":                     backupName,
+			"cassandraDatacenter.name": "dc1",
+		},
+		KubectlOptions: k8s.NewKubectlOptions("", "", namespace),
+	}
+	// Start backup
+	helm.Install(t, helmOptions, backupChartPath, backupName)
+
+	// Wait for the backup to be finished
+	g(t).Eventually(func() int {
+		return backupIsFinished(t, backupName)
+	}, 2*time.Minute, 5*time.Second).Should(gomega.Equal(1))
+}
+
+func backupIsFinished(t *testing.T, backupName string) int {
+	var nodes []string
+	log.Printf("Checking if backup %s is finished...", backupName)
+	kubectlOptions := k8s.NewKubectlOptions("", "", namespace)
+	output, err := k8s.RunKubectlAndGetOutputE(t, kubectlOptions, "get", "cassandrabackup", backupName, "-o", "jsonpath='{.status.finished}'")
+	if err == nil && len(output) > 0 {
+		json.Unmarshal([]byte(strings.ReplaceAll(output, "'", "")), &nodes)
+	}
+	return len(nodes)
+}
+
+func RestoreBackup(t *testing.T, backupName string) {
+	restoreChartPath, err := filepath.Abs("../../charts/restore")
+	g(t).Expect(err).To(gomega.BeNil(), "Couldn't find the absolute path for restore charts")
+
+	helmOptions := &helm.Options{
+		SetValues: map[string]string{
+			"backup.name":              backupName,
+			"cassandraDatacenter.name": "dc1",
+			"name":                     "restore-test",
+		},
+		KubectlOptions: k8s.NewKubectlOptions("", "", namespace),
+	}
+	helm.Install(t, helmOptions, restoreChartPath, "restore-test")
+
+	// Restore should trigger an update of the pod through cass-operator
+	g(t).Eventually(func() string {
+		return readCassandraOperatorProgress(t)
+	}, RETRY_TIMEOUT, 5*time.Second).Should(gomega.Equal("Updating"))
+
+	// Wait for cass-operator to recreate the Cassandra pod
+	g(t).Eventually(func() string {
+		return readCassandraOperatorProgress(t)
+	}, RETRY_TIMEOUT, RETRY_INTERVAL).Should(gomega.Equal("Ready"))
+	WaitForCassDcToBeReady(t)
+}
+
+func readCassandraOperatorProgress(t *testing.T) string {
+	kubectlOptions := k8s.NewKubectlOptions("", "", namespace)
+	cassandraOperatorProgress, err := k8s.RunKubectlAndGetOutputE(t, kubectlOptions, "get", "cassandradatacenter/dc1", "-o", "jsonpath={.status.cassandraOperatorProgress}")
+	g(t).Expect(err).To(gomega.BeNil(), "Couldn't read cass-operator progress state for cassandradatacenter/dc1")
+
+	return cassandraOperatorProgress
+}
+
+// Reaper related steps
+func CheckClusterIsRegisteredInReaper(t *testing.T, clusterName string) {
+	g(t).Eventually(func() bool {
+		return clusterIsRegisteredInReaper(t, clusterName)
+	}, RETRY_TIMEOUT, RETRY_INTERVAL).Should(gomega.BeTrue(), "Cluster wasn't properly registered in Reaper")
+
+	t.Log("Cluster wasn't properly registered in Reaper")
+	t.FailNow()
+}
+
+func clusterIsRegisteredInReaper(t *testing.T, clusterName string) bool {
+	restClient := resty.New()
```

Comment:
I think @burmanm already mentioned https://github.com/k8ssandra/reaper-client-go. It is definitely complete but it is used in reaper-operator and would be nice to reuse it here.

Reply:
Yes, it's planned 👍

Reply:
I'd like to reduce the scope of this PR and the related issue and defer the use of the reaper go client to a later time. It requires to make some changes in the Reaper client to support the endpoints that are needed to trigger a repair, cancel it and check the segments state, which is a little out of scope.
I'll create a ticket to track this.